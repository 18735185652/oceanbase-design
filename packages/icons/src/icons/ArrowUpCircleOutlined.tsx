// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import ArrowUpCircleOutlinedSvg from '../asn/ArrowUpCircleOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const ArrowUpCircleOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={ArrowUpCircleOutlinedSvg} />;

ArrowUpCircleOutlined.displayName = 'ArrowUpCircleOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(ArrowUpCircleOutlined);