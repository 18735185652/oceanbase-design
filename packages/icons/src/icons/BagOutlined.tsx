// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import BagOutlinedSvg from '../asn/BagOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const BagOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={BagOutlinedSvg} />;

BagOutlined.displayName = 'BagOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(BagOutlined);