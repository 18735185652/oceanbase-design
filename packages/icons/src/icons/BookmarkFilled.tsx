// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import BookmarkFilledSvg from '../asn/BookmarkFilled';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const BookmarkFilled = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={BookmarkFilledSvg} />;

BookmarkFilled.displayName = 'BookmarkFilled';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(BookmarkFilled);