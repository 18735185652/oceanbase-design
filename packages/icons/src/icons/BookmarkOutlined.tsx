// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import BookmarkOutlinedSvg from '../asn/BookmarkOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const BookmarkOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={BookmarkOutlinedSvg} />;

BookmarkOutlined.displayName = 'BookmarkOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(BookmarkOutlined);