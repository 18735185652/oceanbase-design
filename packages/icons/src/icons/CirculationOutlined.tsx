// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import CirculationOutlinedSvg from '../asn/CirculationOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const CirculationOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={CirculationOutlinedSvg} />;

CirculationOutlined.displayName = 'CirculationOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(CirculationOutlined);