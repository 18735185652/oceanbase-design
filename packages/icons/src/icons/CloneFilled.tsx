// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import CloneFilledSvg from '../asn/CloneFilled';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const CloneFilled = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={CloneFilledSvg} />;

CloneFilled.displayName = 'CloneFilled';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(CloneFilled);