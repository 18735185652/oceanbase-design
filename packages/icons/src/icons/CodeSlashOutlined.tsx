// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import CodeSlashOutlinedSvg from '../asn/CodeSlashOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const CodeSlashOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={CodeSlashOutlinedSvg} />;

CodeSlashOutlined.displayName = 'CodeSlashOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(CodeSlashOutlined);