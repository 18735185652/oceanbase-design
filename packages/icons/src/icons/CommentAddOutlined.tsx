// DON NOT EDIT IT MANUALLY
import * as React from 'react';
import CommentAddOutlinedSvg from '../asn/CommentAddOutlined';
import AntdIcon, { AntdIconProps } from '../components/AntdIcon';

const CommentAddOutlined = (
  props: AntdIconProps,
  ref: React.ForwardedRef<HTMLSpanElement>,
) => <AntdIcon {...props} ref={ref} icon={CommentAddOutlinedSvg} />;

CommentAddOutlined.displayName = 'CommentAddOutlined';
export default React.forwardRef<HTMLSpanElement, AntdIconProps>(CommentAddOutlined);