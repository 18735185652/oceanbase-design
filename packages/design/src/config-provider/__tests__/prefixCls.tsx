import React from 'react';
import { render } from '@testing-library/react';
import { Button, ConfigProvider } from '@oceanbase/design';
import { PlusOutlined } from '@oceanbase/icons';

describe('ConfigProvider prefixCls', () => {
  it('prefixCls', () => {
    const { container, asFragment } = render(
      <ConfigProvider>
        <Button>button</Button>
        <ConfigProvider prefixCls="customPrefixCls">
          <Button>button</Button>
          <ConfigProvider>
            <Button>button</Button>
          </ConfigProvider>
        </ConfigProvider>
      </ConfigProvider>
    );
    expect(container.querySelectorAll('.ant-btn').length).toBe(1);
    expect(container.querySelectorAll('.customPrefixCls-btn').length).toBe(2);
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  it('iconPrefixCls', () => {
    const { container, asFragment } = render(
      <ConfigProvider>
        <PlusOutlined />
        <ConfigProvider iconPrefixCls="customIconPrefixCls">
          <PlusOutlined />
          <ConfigProvider>
            <PlusOutlined />
          </ConfigProvider>
        </ConfigProvider>
      </ConfigProvider>
    );
    expect(container.querySelectorAll('.anticon').length).toBe(1);
    expect(container.querySelectorAll('.customIconPrefixCls').length).toBe(2);
    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
