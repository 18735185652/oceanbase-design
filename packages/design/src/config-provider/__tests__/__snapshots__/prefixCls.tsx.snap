// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigProvider prefixCls prefixCls 1`] = `
<div
  class="ant-app"
>
  <button
    class="ant-btn ant-btn-default"
    type="button"
  >
    <span>
      button
    </span>
  </button>
  <div
    class="customPrefixCls-app"
  >
    <button
      class="customPrefixCls-btn customPrefixCls-btn-default"
      type="button"
    >
      <span>
        button
      </span>
    </button>
    <div
      class="customPrefixCls-app"
    >
      <button
        class="customPrefixCls-btn customPrefixCls-btn-default"
        type="button"
      >
        <span>
          button
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`ConfigProvider prefixCls iconPrefixCls 1`] = `
<div
  class="ant-app"
>
  <span
    aria-label="plus"
    class="anticon anticon-plus"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="plus"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
      />
      <path
        d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
      />
    </svg>
  </span>
  <div
    class="ant-app"
  >
    <span
      aria-label="plus"
      class="customIconPrefixCls customIconPrefixCls-plus"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </span>
    <div
      class="ant-app"
    >
      <span
        aria-label="plus"
        class="customIconPrefixCls customIconPrefixCls-plus"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
